extensions/scarb-snforge-test-collector/src/crate_collection.rs

```diff
@@ -0,0 +1,82 @@
+use anyhow::{anyhow, Context, Result};
+use camino::{Utf8Path, Utf8PathBuf};
+use serde::Serialize;
+use walkdir::WalkDir;
+
+#[derive(Debug, PartialEq, Clone, Serialize)]
+pub enum CrateLocation {
+    /// Main crate in a package
+    Lib,
+    /// Crate in the `tests/` directory
+    Tests,
+}
+
+#[derive(Debug, PartialEq)]
+pub struct TestCompilationTarget {
+    pub crate_root: Utf8PathBuf,
+    pub crate_name: String,
+    pub crate_location: CrateLocation,
```

Comment:
Is this necessary in the collector at all? We use this information in forge, but it doesn't seem like it is needed here.

Reply:
I mean `crate_location` specifically. This is only useful in forge and only for pretty printing.

Reply:
I don't expect us to make breaking changes in test collection. Otherwise they will be breaking anyways, so I think it's fine to not be backwards compatible then